eyelid-match-ops/src/primitives/poly/test/mul.rs

```diff
@@ -18,11 +18,7 @@ fn test_cyclotomic_mul_rand() {
     }
 
     // Xˆ{N-1}, multiplying by it will rotate by N-1 and negate (except the first).
-    //
-    // Since the degree is less than MAX_POLY_DEGREE, this is already reduced.
-    // It is also in canonical form, because the leading coefficient is non-zero.
-    let mut xnm1 = Poly::zero();
-    xnm1[MAX_POLY_DEGREE - 1] = Coeff::one();
+    let xnm1 = Poly::xn(MAX_POLY_DEGREE - 1);
 
     assert_eq!(xnm1.degree(), MAX_POLY_DEGREE - 1);
 
```

Comment:
Line 33 (right below) is failing with:
> ---- primitives::poly::test::mul::test_cyclotomic_mul_rand stdout ----
> thread 'primitives::poly::test::mul::test_cyclotomic_mul_rand' panicked at eyelid-match- ops/src/primitives/poly/test/mul.rs:33:19:
> accessed virtual leading zero coefficient: improve performance by stopping at the highest non-zero coefficient

But only in the tiny coefficient tests:
https://github.com/Inversed-Tech/eyelid/actions/runs/8639845243/job/23686846926?pr=33#step:5:18

This is probably a zero coefficient at random, but it shouldn't be?

Reply:
Fixed in [c635f4e](https://github.com/Inversed-Tech/eyelid/pull/33/commits/c635f4e8c74b46321e4dc8f0520ba53271c964f1)